lib/modules.nix

```diff
@@ -228,6 +229,28 @@ let
               within a configuration, but can be used in module imports.
             '';
           };
+
+          _module.freeformConfig = mkOption {
+            # Do not show this in the documentation
+            internal = true;
+            visible = false;
+            # Cant add because it has a default
+            readOnly = true;
+            # TODO: Implement the type
```

Comment:
If we can't use a type properly here because of implementation reasons (see `apply`, which is a workaround), we might put a `throw "Type for _module.freeformConfig not implemented. The option value is currently hardcoded."` placeholder here, ideally with a little bit more explanation.

Reply:
Cannot add a throw here so easily, because it breaks the documentation rendering unfortunately.